packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -134,6 +134,7 @@ class UpgradeCommandRunner {
       );
     }
     recordState(flutterVersion);
+    globals.printStatus('Upgrading Flutter to ${upstreamRevision} from ${flutterVersion.frameworkRevision} ($workingDirectory...)');
```

Comment:
We don't currently have any tests for what gets printed during the upgrade. The closest example is one that checks that a message is printed if nothing is upgraded:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart#L92

I would see if you adapt that to add coverage for this.

Reply:
Thanks for reviewing it for you.
I agree with you about adding a test.
I'll TRY to do it.

Reply:
I'm sorry.
I tried, but I can't add the test.
I want to check the standard output, please tell me how to do it.

Reference.
 > expect(testLogger.statusText, contains('Flutter is already up to date'));

I ran the test and the testLogger.statusText was empty.

Please let me know.

Reply:
@karamage check if the status text you want to display is actually being displayed, like these lines in a similar test: 

https://github.com/flutter/flutter/blob/fb508db94d9ecfac3b21f961bfd835f77eca9b88/packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart#L115-L117

I think you should take this (similar)test as a reference to create your new test:

https://github.com/flutter/flutter/blob/fb508db94d9ecfac3b21f961bfd835f77eca9b88/packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart#L310-L314

Reply:
@RoyARG02 Thank you. I'll try.

Reply:
@RoyARG02 I add the test to test/commands.shard/permeable/upgrade_test.dart L310.
But the test fails.

```dart
testUsingContext('Show current version to the upgrade message.', () async {
        final Future<FlutterCommandResult> result = fakeCommandRunner.runCommand(
          force: false,
          continueFlow: false,
          testFlow: false,
          gitTagVersion: gitTagVersion,
          flutterVersion: flutterVersion,
          verifyOnly: false,
        );
        expect(await result, FlutterCommandResult.success());
        expect(testLogger.statusText, contains('Upgrading Flutter to'));
        expect(testLogger.statusText, contains('from ${flutterVersion.frameworkRevision} in'));
        expect(processManager.hasRemainingExpectations, isFalse);
      }, overrides: <Type, Generator>{
        ProcessManager: () => processManager,
        Platform: () => fakePlatform,
      });
```

```
00:05 +12: UpgradeCommandRunner runs upgrade Show current version to the upgrade message.
Expected: contains 'Upgrading Flutter to'
  Actual: ''
```

testLogger.statusText was empty.
Please tell me where I'm wrong.
Thanks for the help.

Reply:
Do you know how to run the test locally? There are some instructions in `packages/flutter_tools/README.md`. You should be able to run this with `dart test test/commands.shard/permeable/upgrade_test.dart`.

Once you have that running, start with something simple like adding `print('here');` to the tool code to make sure it is running what you think it is running

Reply:
I read the README and ran the test locally.

... /... /bin/dart pub run test test test/commands.shard/permeable/upgrade_test.dart

But the test failed.


I will learn how to test using the simple method you have taught me.
Thank you.

Reply:
Try this out for a bit and let me know if you make any progress. I will take a look at this PR again tomorrow and make some suggestions to get your test working correctly

Reply:
@karamage I think you need to use `realCommandRunner` in this case, as `fakeCommandRunner.upgradeChannel` doesn't do anything:
https://github.com/flutter/flutter/blob/fb508db94d9ecfac3b21f961bfd835f77eca9b88/packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart#L401

It also explains you getting **`Actual: ''`** in your test.

Reply:
Thank you. I appreciate it.

Okay, I understand.

I will try to write the test code using realCommandRunner.

Reply:
@RoyARG02 @jonahwilliams 

I've added the test code.

I would like you to review it.

Thank you for showing me how to test it.

The test was a success.

```
../../bin/dart pub run test test/commands.shard/permeable/upgrade_test.dart
00:04 +15: All tests passed!
```

Reply:
🎊